src/v/raft/recovery_stm.cc

```diff
@@ -409,7 +414,11 @@ ss::future<> recovery_stm::replicate(
     // reader
     auto prev_log_idx = model::prev_offset(_base_batch_offset);
     model::term_id prev_log_term;
-
+    if (!_ptr->is_elected_leader()) {
```

Comment:
We check `is_recovery_finished()` before calling `replicate()` (in the same continuation I think) so shouldn't really be necessary? Was the crash in the `install_snapshot` call?

Reply:
yes, in install snapshot

Reply:
i will change the call in install snapshot fiber to the `is_recovery_finished`